lightning/src/ln/script.rs

```diff
@@ -238,10 +238,9 @@ mod shutdown_script_tests {
 
 	#[test]
 	fn generates_segwit_from_non_v0_witness_program() {
-		let version = u5::try_from_u8(16).unwrap();
-		let witness_program = Script::new_witness_program(version, &[0; 40]);
+		let witness_program = Script::new_witness_program(WitnessVersion::V16, &[0; 40]);
 
-		let version = NonZeroU8::new(version.to_u8()).unwrap();
+		let version = NonZeroU8::new(WitnessVersion::V16 as u8).unwrap();
 		let shutdown_script = ShutdownScript::new_witness_program(version, &[0; 40]).unwrap();
```

Comment:
Could we update `new_witness_program` to take `WitnessVersion` instead of `NonZeroU8` in a separate commit?